### PR TITLE
some docs updates to files api

### PR DESCRIPTION
**Pull request by haakonvt**
https://cognitedata.atlassian.net/browse/DM-2393

### CODE REVIEW

File: cognite/client/_api/files.py [hunk 1]
```diff
@@ -685,7 +692,7 @@ def upload_bytes(
     ) -> FileMetadata:
         """Upload bytes or string.
 
-        You can also pass a file handle to content.
+        You can also pass a file handle to 'content'. The file should be opened in binary mode.
```
Comment (tuanng-cognite) [inline cognite/client/_api/files.py:695]: `The file should be opened in binary mode.` I think we don't need to mention this. After all, any content form (str, bytes, TextIO, BinaryIO) is accepted

File: cognite/client/_api/files.py [hunk 2]
```diff
@@ -992,7 +992,7 @@ def _create_unique_file_names(file_names_in: list[str] | list[Path]) -> list[str
 
             file_suffixes = Path(file_name).suffixes
             file_postfix = "".join(file_suffixes)
-            file_base = file_name[: file_name.index(file_postfix) or None]  # Awaiting .removesuffix in 3.9:
+            file_base = file_name.removesuffix(file_postfix)
```
Comment (tuanng-cognite) [inline cognite/client/_api/files.py:995]: nice

File: cognite/client/_api/files.py [hunk 3]
```diff
@@ -627,11 +627,18 @@ def upload_content_bytes(
 
         Examples:
 
-            Upload a file from memory::
+            Finish a file creation by uploading the content using external_id:
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/_api/files.py:630]: Dude, read the next example ❤️

File: cognite/client/_api/files.py [hunk 4]
```diff
@@ -653,7 +660,7 @@ def _upload_bytes(self, content: bytes | TextIO | BinaryIO, returned_file_metada
             full_upload_url = upload_url
         else:
             full_upload_url = urljoin(self._config.base_url, upload_url)
-        file_metadata = FileMetadata.load(returned_file_metadata)
+        file_metadata = FileMetadata._load(returned_file_metadata)
```
[the remark this answers is not in the document]
Comment (haakonvt) [reply cognite/client/_api/files.py:663]: Public that accepts several types vs private that needs exactly an API response. We know what `returned_file_metadata` is, so we use the fast one

File: cognite/client/_api/files.py (hunk 1 above)
Comment (haakonvt) [reply cognite/client/_api/files.py:695]: I disagree, `requests` will stop supporting this in the next major release:

image

**Pull request merged**